app/models/journal/aggregated_journal.rb

```diff
@@ -182,21 +202,31 @@ def table_name
            :id,
            :version,
            :attributes,
+           :attachable_journals,
+           :customizable_journals,
+           :editable_by?,
            to: :journal
 
   def initialize(journal)
     @journal = journal
   end
 
+  # returns an instance of this class that is reloaded from the database
+  def reloaded
+    self.class.with_notes_id(notes_id)
+  end
+
   def user
     @user ||= User.find(user_id)
   end
 
   def predecessor
     unless defined? @predecessor
+      version_sql = "COALESCE(addition.version, #{Journal.table_name}.version)"
       raw_journal = self.class.query_aggregated_journals(journable: journable)
-                    .where("#{Journal.table_name}.version < ?", version)
-                    .order("#{Journal.table_name}.version DESC")
+                    .where("#{version_sql} < ?", version)
+                    .except(:order)
```

Comment:
Given the fact that orders are _not_ "overwritten" this makes actually sense to me now. Thanks!